atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/ThriftObjectSizeUtils.java

```diff
@@ -0,0 +1,229 @@
+                + Integer.BYTES
+                + getCqlMetadataSize(cqlResult.getSchema());
+    }
+
+    public static long getKeySliceSize(KeySlice keySlice) {
+        if (keySlice == null) {
+            return getNullSize();
+        }
+
+        return getByteArraySize(keySlice.getKey())
+                + getCollectionSize(keySlice.getColumns(), ThriftObjectSizeUtils::getColumnOrSuperColumnSize);
+    }
+
+    public static long getStringSize(String string) {
+        if (string == null) {
+            return getNullSize();
+        }
+
+        return string.length() * Character.SIZE;
+    }
+
+    private static long getCounterSuperColumnSize(CounterSuperColumn counterSuperColumn) {
+        if (counterSuperColumn == null) {
+            return getNullSize();
+        }
+
+        return getByteArraySize(counterSuperColumn.getName())
+                + getCollectionSize(counterSuperColumn.getColumns(), ThriftObjectSizeUtils::getCounterColumnSize);
+    }
+
+    private static long getCounterColumnSize(CounterColumn counterColumn) {
+        if (counterColumn == null) {
+            return getNullSize();
+        }
+
+        return getByteArraySize(counterColumn.getName()) + getCounterValueSize();
+    }
+
+    private static long getSuperColumnSize(SuperColumn superColumn) {
+        if (superColumn == null) {
+            return getNullSize();
+        }
+
+        return getByteArraySize(superColumn.getName())
+                + getCollectionSize(superColumn.getColumns(), ThriftObjectSizeUtils::getColumnSize);
+    }
+
+
+    private static long getColumnSize(Column column) {
+        if (column == null) {
+            return getNullSize();
+        }
+
+        return getByteArraySize(column.getValue())
+                + getByteArraySize(column.getName())
+                + getTtlSize()
+                + getTimestampSize();
+    }
+
+    private static long getDeletionSize(Deletion deletion) {
+        if (deletion == null) {
+            return getNullSize();
+        }
+
+        return getTimestampSize()
+                + getByteArraySize(deletion.getSuper_column())
+                + getSlicePredicateSize(deletion.getPredicate());
+    }
+
+    private static long getSlicePredicateSize(SlicePredicate predicate) {
+        if (predicate == null) {
+            return getNullSize();
+        }
+
+        return getCollectionSize(predicate.getColumn_names(), ThriftObjectSizeUtils::getByteBufferSize) + getSliceRangeSize(predicate.getSlice_range());
+    }
+
+    private static long getSliceRangeSize(SliceRange sliceRange) {
+        if (sliceRange == null) {
+            return getNullSize();
+        }
+
+        return getByteArraySize(sliceRange.getStart())
+                + getByteArraySize(sliceRange.getFinish())
+                + getReversedBooleanSize()
+                + getSliceRangeCountSize();
+    }
+
+    private static long getCqlMetadataSize(CqlMetadata schema) {
```

Comment:
Good to see we're being thorough here. However some of these sizes could probably be removed for simplicity. I think we are mainly interested in the sizes of the column names and values we read and write, and not so much the extra metadata included in the request/response objects. (Fine to merge as is for now though)

Reply:
Merging as is for now, with most code paths tested.